library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -84,20 +84,31 @@ class ProtocolClient(
                             finalResponse.trailers,
                         ),
                     )
-                } else {
-                    val responseCodec = serializationStrategy.codec(methodSpec.responseClass)
-                    val responseMessage = responseCodec.deserialize(
-                        finalResponse.message,
-                    )
+                    return@httpClientUnary
+                }
+                val responseCodec = serializationStrategy.codec(methodSpec.responseClass)
+                val responseMessage: Output
+                try {
+                    responseMessage = responseCodec.deserialize(finalResponse.message)
```

Comment:
We were missing exception handling around deserializing using the Codec, which lead to not notifying callers of a failure.